--- a/net/minecraft/pathfinding/PathNodeType.java
+++ b/net/minecraft/pathfinding/PathNodeType.java
@@ -1,6 +1,6 @@
 package net.minecraft.pathfinding;
 
-public enum PathNodeType {
+public enum PathNodeType implements net.minecraftforge.common.IExtensibleEnum {
    BLOCKED(-1.0F),
    OPEN(0.0F),
    WALKABLE(0.0F),
@@ -32,6 +32,10 @@
       this.field_186307_r = p_i46653_3_;
    }
 
+   public static PathNodeType create(String name, float defaultPriority) {
+      throw new IllegalArgumentException("Enum not extended");
+   }
+
    public float func_186289_a() {
       return this.field_186307_r;
    }
